diff --git a/lua/treesitter-matchup/third-party/query.lua b/lua/treesitter-matchup/third-party/query.lua
index add114c..5c8960a 100644
--- a/lua/treesitter-matchup/third-party/query.lua
+++ b/lua/treesitter-matchup/third-party/query.lua
@@ -205,7 +205,7 @@ function M.iter_prepared_matches(query, qnode, bufnr, start_row, end_row)
     curr_obj[path[#path]] = value
   end
 
-  local matches = query:iter_matches(qnode, bufnr, start_row, end_row)
+  local matches = query:iter_matches(qnode, bufnr, start_row, end_row, { all = false })
 
   local function iterator()
     local pattern, match, metadata = matches()
