diff --git a/autoload/highlighturl.vim b/autoload/highlighturl.vim
index c93ae31..3c843e2 100644
--- a/autoload/highlighturl.vim
+++ b/autoload/highlighturl.vim
@@ -16,10 +16,10 @@ endfunction
 
 function! highlighturl#default_pattern() abort
   return  '\v\c%(%(h?ttps?|ftp|file|ssh|git)://|[a-z]+[@][a-z]+[.][a-z]+:)%('
-        \.'[&:#*@~%_\-=?!+;/0-9a-z]+%(%([.;/?]|[.][.]+)[&:#*@~%_\-=?!+/0-9a-z]+|:\d+|'
+        \.'[&:#*@~%_\-–=?!+;/0-9a-z]+%(%([.;/?]|[.][.]+)[&:#*@~%_\-–=?!+/0-9a-z]+|:\d+|'
         \.',%(%(%(h?ttps?|ftp|file|ssh|git)://|[a-z]+[@][a-z]+[.][a-z]+:)@![0-9a-z]+))*|'
-        \.'\([&:#*@~%_\-=?!+;/.0-9a-z]*\)|\[[&:#*@~%_\-=?!+;/.0-9a-z]*\]|'
-        \.'\{%([&:#*@~%_\-=?!+;/.0-9a-z]*|\{[&:#*@~%_\-=?!+;/.0-9a-z]*\})\})+'
+        \.'\([&:#*@~%_\-–=?!+;/.0-9a-z]*\)|\[[&:#*@~%_\-–=?!+;/.0-9a-z]*\]|'
+        \.'\{%([&:#*@~%_\-–=?!+;/.0-9a-z]*|\{[&:#*@~%_\-–=?!+;/.0-9a-z]*\})\})+'
 endfunction
 
 function! highlighturl#get_url_highlight(cursor) abort
